File: Assets/Scripts/SwordOwner.js
#pragma strict

public var distance = 10.0;
private var worldObject : GameObject;
private var player : GameObject;
public var x = 0.42;
public var y = 0.66;
public var z = 0.42;
public var scalef = 8.0;
private var grabbed = false;

function Start() {
	worldObject = GameObject.Find("World");
	player = GameObject.FindWithTag("Player"); // pour trouver le personnage
}

function Update() {
	//if(!grabbed) {
		distance = Vector3.Distance(this.transform.position, player.transform.position);
		Debug.Log("Distance is " + distance);
		if(distance < 1.3) {
			Debug.Log("Grab your destiny!");
			worldObject.SendMessage("CancelTimer");
			this.transform.parent = player.transform;
			if(!grabbed) {
			this.transform.localScale -= new Vector3(scalef, scalef, scalef); grabbed = true;}
			this.transform.localPosition = new Vector3(x, y, z);
			this.transform.localRotation = Quaternion.identity;
			this.transform.localRotation = Quaternion.Euler(90, 0, -120);
			//grabbed = true;
		}
	//}
}

File: Assets/Scripts/SwordOwner.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class SwordOwner : MonoBehaviour {
	public float distance = 10.0f;
	private GameObject worldObject;
	private GameObject player;
	public float x = 0.42f;
	public float y = 0.66f;
	public float z = 0.42f;
	public float scalef = 8.0f;

	void Start() {
		worldObject = GameObject.Find("World");
		player = GameObject.FindWithTag("Player"); // pour trouver le personnage
		if(GameVariables.swordGrabbed) {
			GrabSword();
		}
	}

	void Update() {
		if(!GameVariables.swordGrabbed) {
			distance = Vector3.Distance(this.transform.position, player.transform.position);
			//Debug.Log("Distance is " + distance);
			if(distance < 1.3) {
				GrabSword();
			}
		}
	}

	void GrabSword() {
		Debug.Log("Grab your destiny!");
		worldObject.SendMessage("CancelTimer");
		this.transform.parent = player.transform;
		this.transform.localScale -= new Vector3(scalef, scalef, scalef);
		this.transform.localPosition = new Vector3(x, y, z);
		this.transform.localRotation = Quaternion.identity;
		this.transform.localRotation = Quaternion.Euler(90, 0, -120);
		GameVariables.swordGrabbed = true;
	}
}